### PR TITLE
[jsk_apc2016_common][jsk_2016_01_baxter_apc] Segmentation in bin may fes

### PR DESCRIPTION
**Pull request by yuyu2172**
## related issue
#1457 

## contents
1. #1465, #1466, #1467, #1468 
2. rviz config
3. changed launch file to visualize 
4. ~~bug fix for self filters written by @knorth55~~

### CODE REVIEW

File: jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py [hunk 1]
```diff
@@ -72,11 +74,11 @@ def _callback(self, sync):
         self.set_apc_sample()
         # generate a binary image
         self.segmentation()
+
```
Comment (knorth55) [inline jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py]: not needed change

File: jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py [hunk 2]
```diff
@@ -72,11 +74,11 @@ def _callback(self, sync):
...
         try:
             predict_msg = self.bridge.cv2_to_imgmsg(
                     self.predicted_segment, encoding="mono8")
             predict_msg.header = color_img.header
-
```
Comment (knorth55) [inline jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py]: not needed change

**Comment by knorth55** [comment]
https://github.com/start-jsk/jsk_apc/pull/1469/commits/fa6a2268b20c68f92912e3aeb30330680d5b797b
it is totally different from commit message

**Comment by knorth55** [comment]
https://github.com/start-jsk/jsk_apc/pull/1469/commits/bdf4e7c5a7c04fb6ab4722f3b8c999a784ebd3eb can be squashed

**Comment by knorth55** [comment]
rebase it

**Comment by knorth55** [comment]
main.launch should not be added

File: jsk_2016_01_baxter_apc/launch/segmentation_in_bin.launch [hunk 3]
```diff
@@ -1,5 +1,7 @@
 <launch>
   <arg name="TRAINED_PATH" value="$(find jsk_apc2016_common)/data/trained_segmenter.pkl" />
+  <arg name="INPUT_BIN_ARRAY" value="/set_bin_param/bin_array"/>
```
Comment (knorth55) [inline jsk_2016_01_baxter_apc/launch/segmentation_in_bin.launch]: `set_bin_param` node is started in this launch file, and it depends on node name in thisfile.
thereforem we don't need to modify `arg` param.

File: jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py [hunk 4]
```diff
@@ -31,38 +30,36 @@ def __init__(self):
...
-                queue_size=50,
-                slop=1.0)
-        self.sync.registerCallback(self._callback)
+        self.subscriber = rospy.Subscriber('~input', SegmentationInBinSync, self._callback)
 
     def unsubscribe(self):
         self.sub.unregister()
 
-    def _callback(self, dist_msg, height_msg, img_msg, mask_msg):
+    def _callback(self, sync):
```
Comment (knorth55) [inline jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py]: `sync` -> `sync_msg`

File: jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py [hunk 5]
```diff
@@ -31,38 +30,36 @@ def __init__(self):
...
         rospy.loginfo('started')
-        self.height = dist_msg.height
-        self.width = dist_msg.width
+
+        dist_img = sync.dist_msg
+        height_img = sync.height_msg
+        color_img = sync.color_msg
+        mask_img = sync.mask_msg
```
Comment (knorth55) [inline jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py]: `dist_img` -> `dist_msg`

File: jsk_apc2016_common/src/sib_topic_synchronizer.cpp [hunk 6]
```diff
@@ -1,62 +1,49 @@
...
-  message_filters::Subscriber<CameraInfo> info_sub(nh, "input/info", 1);
-  message_filters::Subscriber<PointCloud2> point_sub(nh, "input", 1);
+    ros::NodeHandle nh("~");
 
-  typedef sync_policies::ApproximateTime<Image, CameraInfo, PointCloud2> MySyncPolicy;
-  Synchronizer<MySyncPolicy> sync(MySyncPolicy(100), image_sub, info_sub, point_sub);
+    pub_ = nh.advertise<jsk_apc2016_common::SegmentationInBinSync>("output", 1);
 
-  sync.registerCallback(boost::bind(&callback, _1, _2, _3));
+    message_filters::Subscriber<Image> image_sub(nh, "input/image", /*queue_size*/1);
```
Comment (knorth55) [inline jsk_apc2016_common/src/sib_topic_synchronizer.cpp]: `queue_size` is not correct, it is `callback_queue`
i think we dont need this comment out

File: jsk_apc2016_common/src/sib_topic_synchronizer.cpp [hunk 7]
```diff
@@ -1,62 +1,49 @@
...
-    points_sent = (PointCloud2) *points;
-
     jsk_apc2016_common::SegmentationInBinSync sync_data;
-    sync_data.image_color = image_sent;
-    sync_data.cam_info = caminfo_sent;
-    sync_data.points = points_sent;
+    sync_data.color_msg = *color_msg;
+    sync_data.dist_msg = *dist_msg;
+    sync_data.height_msg = *height_msg;
+    sync_data.mask_msg = *mask_msg;
```
Comment (knorth55) [inline jsk_apc2016_common/src/sib_topic_synchronizer.cpp]: why not `sync_data` -> `sync_msg` ?

**Comment by yuyu2172** [comment]
fixed

**Comment by knorth55** [comment]
i thought when you rebase origin/master, cherry-picked commit is merged, but actually it's not.
can you rebase origin/master and delete cherry-picked commits about self-filter?

**Comment by knorth55** [comment]
+1

**Comment by yuyu2172** [comment]
done

**Comment by yuyu2172** [comment]
@knorth55 
ping

**Comment by yuyu2172** [comment]
@wkentaro 
ready to be merged.

**Pull request merged**